### ISSUE TITLE
Can't reap unbonded validator control account, can't find what extrinsic is holding it up...

### ISSUE THREAD

**Issue by polkadotbabe** [paritytech/polkadot#2654]
Similar to this issue (closed in substrate as not a substrate issue): https://github.com/paritytech/subport/issues/96

After unbonding a validator control/stash pair, waiting the 28 days, "unlocking" in polkadot-JS, the control account for a (former) validator can't reap: "There is an existing reference count on the sender account. As such the account cannot be reaped from the state."

I can re-assign it as a controller to another validator account, and the stash account is able to reap. Purged the keys, called force withdraw... Obviously the calls are on chain, and there's an entire discord log of trouble-shooting steps that didn't work. I'm not concerned about the address being public, not sure if the etiquette here allows posting it directly? 

It seems the JS GUI isn't reading all extrinsics or outright missing one that's preventing the account from reaping...

**Comment by thiolliere** [comment]
so you called  `purge_keys` with the controller and it was successful.
Then you called `unbond(all_stash_active_balance)` with the controller and it was successful.
Then after 28 days you called `withdraw_unbonded` with the controller and it was failing ?

Can you say which extrinsic was failing ?

If you don't mind sharing the account address, I think you can post it to ease understanding the issue.

**Comment by polkadotbabe** [comment]
Calling withdrawUnbonded returns the "staking not controller" error:

https://polkadot.subscan.io/account/1Kbabe7YGBxMYsbMTuW7ga5ctSsh6kcirkvVSTLE1eVAsYB

So, obviously, the address is 1Kbabe7YGBxMYsbMTuW7ga5ctSsh6kcirkvVSTLE1eVAsYB

**Comment by thiolliere** [comment]
The call to purge_keys has failed, thus there is still associated to the old stash account. And this is preventing the validator account to be killed.

You need:
* call `bond` with the old stash, to the controller
* call `purge_keys` with the controller
* call `unbond` with the controller
* wait 28 days
* call `withdraw_unbonded` with the controller

**Comment by polkadotbabe** [comment]
I'll doublecheck (and edit this accordingly) but I already did all that, although I unbonded then purged the keys, the 28 days lapsed, can't withdraw_unbonded....

Were it still bonded to a stash I wouldn't have been able to assign it as controller of another validator's stash?

**Comment by thiolliere** [comment]
I can't see the successful purge_keys in the UI, you did use the same stash and controller isn't it?

You need to use the same stash and controller as when you set the keys.

**Comment by polkadotbabe** [comment]
It was separate stash/controller, digging back through to try and figure out which one... although I did reset and remove it as controller for various accounts....

**Comment by thiolliere** [comment]
> It was separate stash/controller, 

Yes I meant the controller account and the stash account that was in used when you called `set_keys`, should be the same when you call `purge_keys`.

If `purge_keys` failed you need to bond you stash to your controller and call purge_keys.
You can do the step as described in https://github.com/paritytech/polkadot/issues/2654#issuecomment-804273980

**Comment by polkadotbabe** [comment]
I still can't reap this -- despite the js ui showing that the coins are "available", I can't transfer them out.

I'm not sure if I was clear above, I simply unbonded them per the instructions. It worked on one bond, not on this one. I've called purge_keys, tried to force the withdraw, nothing in any extrinsic works. 

As a separate issue with two separate accounts, my nomination stash can't unbond, I'll post screenshots -- when I click on the unlock icon it says -- despite 1 DOT being free in the account -- there's not enough funds to unbond.

These ongoing errors in published code are intensely frustrating, given I've had a decent chuck on money sitting there locked via errors, I'd really appreciate DOT assigning someone to solve this. 

Thanks in advance, and have a great day!

--Anna

**Comment by polkadotbabe** [comment]
![DOT_cantsent2](https://user-images.githubusercontent.com/79539755/134552914-929747c0-4681-4f44-830a-c0b24b8e855f.jpg)
![DOT_cantsent1](https://user-images.githubusercontent.com/79539755/134552918-4563a4bb-4694-46ce-b767-e454074ffa60.jpg)

In these, you can see the above account link, it says available but it's not. Somehow while waiting it cleared 0.98 or so, there used to be 1 DOT stuck. If you review the TXes I've called all the required extrinsics with no success.

**Comment by polkadotbabe** [comment]
![Dot3](https://user-images.githubusercontent.com/79539755/134553113-12ed0e55-7ddc-4ac1-bdb3-864a07f84f2e.png)

In this I'm not sure what it's saying -- my addresses are all vanity beginning with 1babe or 12babe and aren't on the list as I didn't nominate them -- but it seems Allnodes #1 complained about W3F which I did nominate, and now DOT wants me to pay 1.0394 for an anonymous validators' slashing? Which might be why the following can't unbond:

**Comment by polkadotbabe** [comment]
![Dot2_a](https://user-images.githubusercontent.com/79539755/134553940-bd459d25-836b-4256-a262-87bb6c8a03d2.jpg)
![Dot2](https://user-images.githubusercontent.com/79539755/134553944-9bc65e05-a081-440e-8db8-08298de73683.png)
![DOT_Cant_Unlock](https://user-images.githubusercontent.com/79539755/134553945-952ac1a0-94f5-4345-8cbe-a682dcabe4da.png)

In these, you can see trying to unlock returns an error of lacking funds, even though there's 0.9877 free in the stash account and 0.418 in the controller accounts.

Further, I can't "Stop" the bond as pulling down the three-dot menu to choose unbond funds results in no action.

Thanks again for helping DOT/Parity!

**Comment by polkadotbabe** [comment]
oh, nominator control: 
1babenhvYgxX7x875dXfQMRbmxEqXTW6kcSAsyN3CkU8cNw

Nominator stash: 1babeEzMQKnbcpTi3iqfLXQCkTKJV5ufuAqp7zAtCmRYCcB

**Comment by bkchr** [comment]
CC @rrtti

**Comment by rrtti** [comment]
@michalisFr is looking into this.
